### PR TITLE
typhoon_h480: rotate sonar such that is forward facing

### PR DESCRIPTION
**Pull request by mrivi**
The sonar was backward facing which doesn't make much sense. I changed it back to forward facing

### CODE REVIEW

**Comment by TSC21** [comment]
@mrivi actually this needs to be revisited. When https://github.com/PX4/sitl_gazebo/pull/339 was applied, The 1.57 rotation that you see meant that the sonar was downward facing - not backwards. At least from my tests, the rotation reported by Mavlink was consistent with the rotation of the beam (maybe not the visual of it).
Can you please confirm that the sonar link rotation is consistent on models using the sonar or the mb1240-xl-ez4? We need to make sure that the sensor faces forward, but with a 1.57 rotation, that the beam faces down and the Mavlink orientation matches with the actual expected sensor orientation? (Ex: pose of `0 0 0 0 1.57 0` of the sonar with respect to the `base_link`, should result on a sensor orientation of [24](https://mavlink.io/en/messages/common.html#MAV_SENSOR_ROTATION_PITCH_90)).

**Comment by TSC21** [comment]
Also, the sonar sensor link and the lidar sensor links behave differently, as it can be seen here: https://github.com/PX4/sitl_gazebo/pull/339/files#diff-1155e8d444b5184de0a4b9e4d0aa4cb6R248. So this need to to be treated in a way we keep the implementation consistent and that the orientation report stays correct.

File: models/sonar/model.sdf [hunk 1]
```diff
@@ -28,7 +28,7 @@
         </material>
       </visual>
       <sensor name="sonar" type="sonar">
-        <pose>0 0 0 0 -1.57 0</pose>
```
Comment (TSC21) [inline models/sonar/model.sdf:31]: Actually, this wasn't forward facing before: https://github.com/PX4/sitl_gazebo/pull/339/files#diff-d82a6bf200434ac922abd2c2712fe1f6L28

Comment (mrivi) [reply models/sonar/model.sdf:31]: No, originally it was forward facing. 100% sure of this.

Comment (mrivi) [reply models/sonar/model.sdf:31]: https://github.com/PX4/sitl_gazebo/pull/339/files#diff-1155e8d444b5184de0a4b9e4d0aa4cb6R248 if you set the `</pose>` as described here, the reported orientation is 0 (forward facing) but the sonar is backward facing!

Comment (TSC21) [reply models/sonar/model.sdf:31]: Doesn't matter now. The orientation logic required some changes on the models and sensors. Please make sure your change still guarantees that the protobuf Range msg orientation and the distance_sensor msg orientation fields make sense.

Comment (TSC21) [reply models/sonar/model.sdf:31]: > https://github.com/PX4/sitl_gazebo/pull/339/files#diff-1155e8d444b5184de0a4b9e4d0aa4cb6R248 if you set the `</pose>` as described here, the reported orientation is 0 (forward facing) but the sonar is backward facing!

If you take a look at the source code at master, that function is not there anymore.

**Comment by TSC21** [comment]
@mrivi what matters in terms of the orientation is that the final orientation that is reported on the Mavlink message stays correct. If the the reported pose is correct and the sensor beam points at the correct direction, then the change should be fine. It doesn't matter if the sensor link has X pointing forward or backwards.

**Comment by Jaeyoung-Lim** [comment]
Any updates on this? the h480 sonar is still pointing in the wrong way...

**Comment by mrivi** [comment]
@TSC21 - It seems that the orientation is correct. Orientation 0 is  ROTATION_FORWARD_FACING


typhoon_sonar

**Review by TSC21 — APPROVED** [review]
Great! Merging

**Pull request merged**